### PR TITLE
BMU-401 fix(sync-actions): fix type sync-actions errors

### PR DESCRIPTION
**Pull request by markus-azer**
#### Todo

- Tests
  - [ ] Unit
  - [ ] Integration
  - [ ] Acceptance
- [ ] Documentation
- [ ] `Type` label for the PR

### CODE REVIEW

File: packages/sync-actions/src/types-actions.js [hunk 1]
```diff
@@ -136,7 +136,7 @@ export function actionsMapFieldDefinitions(
           label: extractedPairs.newObj.label,
           fieldName: extractedPairs.oldObj.name,
         })
-      } else if (diffValue.type.values) {
+      } else if (diffValue?.type?.values) {
```
Comment (markus-azer) [inline packages/sync-actions/src/types-actions.js:139]: That one fixes this issue https://github.com/commercetools/nodejs/issues/1766

File: packages/sync-actions/src/types-actions.js [hunk 2]
```diff
@@ -152,7 +152,7 @@ export function actionsMapFieldDefinitions(
         if (diffValue.length === 3 && diffValue[2] === 3) {
           actions.push({
             action: 'changeFieldDefinitionOrder',
-            fieldNames: next,
+            fieldNames: next.map((n) => n.name),
```
Comment (markus-azer) [inline packages/sync-actions/src/types-actions.js:155]: doc: https://docs.commercetools.com/api/projects/types#change-the-order-of-fielddefinitions

File: packages/sync-actions/src/types-actions.js (hunk 1 above)
Comment (nicolasnieto92) [reply packages/sync-actions/src/types-actions.js:139]: good catch! although this is not throwing error anymore, i think the `required` field is still not being updated.
so [this](https://github.com/commercetools/commercetools-importer/pull/2552/files#diff-b9adf055e61d9e3c62b4ffb433f171742bd5f8b04d226bc3919c0c257d476ac5R75-R97) particular test is not failing anymore, but if we check the `required` field, its still `false` if im not mistaken.

File: packages/sync-actions/test/types-sync-fields.spec.js [hunk 3]
```diff
@@ -336,4 +336,28 @@ describe('Actions', () => {
...
+            name: 'first',
+            required: true,
+          },
+        ],
+      })
+      now = createTestType({
+        fieldDefinitions: [
+          {
+            name: 'first',
+            required: false,
```
Comment (tdeekens) [inline packages/sync-actions/test/types-sync-fields.spec.js:353]: Sorry, I am not following: we change the value but don't create an update action. Can you elaborate?

File: packages/sync-actions/src/types-actions.js (hunk 1 above)
Comment (markus-azer) [reply packages/sync-actions/src/types-actions.js:139]: Yep very true, this field is unchangeable, so the current behaviour is just ignore these fields

File: packages/sync-actions/test/types-sync-fields.spec.js (hunk 3 above)
Comment (markus-azer) [reply packages/sync-actions/test/types-sync-fields.spec.js:353]: totally agree the test is already confusing 👍 (gonna rename it better and add more context to test), there is some fields that is immutable or unchangeable so there is no `update actions` for it, which apply for the this field `FieldDefinition->required` (maybe also that should be mentioned in the public doc cause there is no clue cause it also apply for `resourceTypeIds`, `FieldDefinition->Name`).

and the current behaviour is to just ignore it for example in product entity , `[productType](https://docs.commercetools.com/api/projects/products#product)`  field `Cannot be changed.`

for above field `required`, changing it causing throwing errors 😞, and thought of adding test for it.

already been thinking about adding util for that case and pass it as config option, so it doesnt require major upgrade.

does it make sense?

Comment (tdeekens) [reply packages/sync-actions/test/types-sync-fields.spec.js:353]: Yeah thanks makes sense.

File: packages/sync-actions/src/types-actions.js (hunk 1 above)
Comment (nicolasnieto92) [reply packages/sync-actions/src/types-actions.js:139]: oh i see! good then 👍

**Review by nicolasnieto92 — APPROVED** [review]
looks good! thanks 🚀

**Review by tdeekens — APPROVED** [review]

**Pull request merged**